airbyte-commons-worker/src/test/java/io/airbyte/workers/general/ReplicationWorkerPerformanceTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.general;
+
+import io.airbyte.config.JobSyncConfig.NamespaceDefinitionType;
+import io.airbyte.config.ReplicationOutput;
+import io.airbyte.config.StandardSyncInput;
+import io.airbyte.metrics.lib.NotImplementedMetricClient;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.CatalogHelpers;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.protocol.models.JsonSchemaType;
+import io.airbyte.protocol.models.SyncMode;
+import io.airbyte.workers.RecordSchemaValidator;
+import io.airbyte.workers.WorkerMetricReporter;
+import io.airbyte.workers.exception.WorkerException;
+import io.airbyte.workers.internal.NamespacingMapper;
+import io.airbyte.workers.internal.book_keeping.AirbyteMessageTracker;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import lombok.extern.slf4j.Slf4j;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Warmup;
+
+@Slf4j
+public class ReplicationWorkerPerformanceTest {
+
+  /**
+   * Hook up the DefaultReplicationWorker to a test harness with an insanely quick Source
+   * {@link LimitedAirbyteSource} and Destination {@link EmptyAirbyteDestination}.
+   * <p>
+   * Harness uses Java Micro Benchmark to run the E2E sync a configured number of times. It then
+   * reports a time distribution for the time taken to run the E2E sync.
+   * <p>
+   * Because the reported time does not explicitly include throughput numbers, throughput logging has
+   * been added. This class is intended to help devs understand the impact of changes on throughput.
+   * <p>
+   * To run this, simply run the main method and run the logs.
+   */
+  @Benchmark
+  // SampleTime = the time taken to run the benchmarked method. Use this because we only care about
+  // the time taken to sync the entire dataset.
+  @BenchmarkMode(Mode.SampleTime)
```

Comment:
@colesnodgrass unfortunately because these annotations don't have the target annotation type, we cannot fold them into a meta annotation. I think this is fine for now.